src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java

```diff
@@ -544,7 +546,12 @@ private void indexFile(FileAbstractModel fileAbstractModel, ScanStatistic stats,
             try {
                 // Create the Doc object
                 Doc doc = new Doc();
-
+                String docId;
+                if (fsSettings.getFs().isFilenameAsId()) {
```

Comment:
`generateIdFromFilename` does already this test I believe. This change is not needed anymore I think.